core/src/test/java/com/graphhopper/routing/util/AbstractBikeFlagEncoderTester.java

```diff
@@ -410,6 +410,24 @@ public void testBarrierAccess() {
         node.setTag("bicycle", "yes");
         // no barrier!
         assertTrue(encoder.handleNodeTags(node) == 0);
+
+        // Issue 407
+        // kissing_gate without bicycle tag
+        node.setTag("barrier", "kissing_gate");
+        // barrier!
+        assertFalse(encoder.handleNodeTags(node) == 0);
```

Comment:
Can you have a look - this test seems to fail for mtb and racingbike according to travis: https://travis-ci.org/github/graphhopper/graphhopper/jobs/774275860
Also down on line 425 for bike and bike2

Reply:
I'm sorry for the inconvenience. I had accidentially disabled the tests and therefore didn't notice :disappointed:.  bba2e67 is a bit more complex now, but does the intended blocking of `barrier=kissing_gate` without `bicycle=yes` except for the mountainbike encoder, where  this is allowed.

Reply:
Hmmh, this is indeed a bit involved. As we run `blockByDefault==false` in production (see our reasoning for this [here](https://forum.openstreetmap.org/viewtopic.php?id=58238)) would it make the code easier if that is enabled in tests too?

Reply:
Sorry, I'm a bit confused now. `blockByDefault` is false for all bicycle encoders, also for the tests. So I don't see how setting it to false once more would simplify the code.
The only change which would simplify the code (and the tests as well) is allowing of a barrier `kissing_gate` per default, without the addition of `bicycle=yes`. But I don't like to do this.

Reply:
The currently handling with the blockByDefault is already not that easy to understand and introducing another concept with different defaults makes it harder to reason about the behaviour. Also having the 3rd absoluteBarriers list is a bit inflexible as tagging cannot influence access.

My proposal: create another pull request where we only have two lists:

 * one is collecting all barriers with a pass-by-default behaviour (passByDefaultBarriers) and 
 * one with a block-by-default behaviour (blockByDefaultBarriers) (basically this is the previous absoluteBarriers)

Would this make sense to you too? 

There or in another issue we should also think about removing the blockByDefault config parameter. E.g. in production we already use `block_barriers=false` even for car.

Reply:
Thank you, this makes sense! I have started with the proposed PR, but I'm facing a couple of car specific high level tests which are failing now. I'll need some help to fix some of them. I'll comment and mark these if this is OK.

Reply:
Sure, will do - thank you!